[SPARK-5013] [MLlib] Added documentation and sample data file for GaussianMixture

tgaloppo opened the pull request:
Simple description and code samples (and sample data) for GaussianMixture

tgaloppo committed:
[SPARK-5013] Added documentation and sample data file for GaussianMixture

SparkQA commented:
[Test build #26845 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/26845/consoleFull) for   PR 4401 at commit [`3eb41fa`](https://github.com/apache/spark/commit/3eb41fa1f304d817109d5f56349dfcca49119957).
 * This patch merges cleanly.

SparkQA commented:
[Test build #26845 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/26845/consoleFull) for   PR 4401 at commit [`3eb41fa`](https://github.com/apache/spark/commit/3eb41fa1f304d817109d5f56349dfcca49119957).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `public class GaussianMixtureExample `

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/26845/
Test PASSed.

mengxr commented on docs/mllib-clustering.md:
```diff
@@ -168,6 +187,112 @@ print("Within Set Sum of Squared Error = " + str(WSSSE))
+parsedData = data.map(lambda line: array([float(x) for x in line.strip().split(' ')]))
+
+# Build the model (cluster the data)
+gmm = new GaussianMixture.train(parsedData, 2)
```
remove `new`

mengxr commented on docs/mllib-clustering.md:
```diff
@@ -168,6 +187,112 @@ print("Within Set Sum of Squared Error = " + str(WSSSE))
+
+<div data-lang="python" markdown="1">
+In the following example after loading and parsing data, we use a
+[`GaussianMixture`](api/scala/index.html#org.apache.spark.mllib.clustering.GaussianMixture) 
```
This should be linked to the Python doc, e.g., `[NaiveBayes](api/python/pyspark.mllib.classification.NaiveBayes-class.html)`.

mengxr commented:
@tgaloppo Thanks for working on the user guide! I saw this is still marked as `WIP`. Are you going to add more content?

tgaloppo commented:
@mengxr I was considering adding a graphic of the test data with the recovered 2-d gaussians... I'm not sure if it would really be beneficial or not.

tgaloppo committed:
Minor fixes

tgaloppo commented:
@mengxr I have made the fixes you pointed out.  I am having trouble building the API docs so I can not verify that the link to the python GaussianMixture class resolves properly.

SparkQA commented:
[Test build #26884 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/26884/consoleFull) for   PR 4401 at commit [`2368690`](https://github.com/apache/spark/commit/23686904dbea03d10264b00b9fffa797d21d43f0).
 * This patch merges cleanly.

SparkQA commented:
[Test build #26884 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/26884/consoleFull) for   PR 4401 at commit [`2368690`](https://github.com/apache/spark/commit/23686904dbea03d10264b00b9fffa797d21d43f0).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `public class GaussianMixtureExample `

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/26884/
Test PASSed.

mengxr commented:
Did you follow the `docs/README.md`?

tgaloppo committed:
Fixed link in mllib-clustering.md
Added Gaussian mixture and power iteration as available clustering techniques in mllib-guide

SparkQA commented:
[Test build #26915 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/26915/consoleFull) for   PR 4401 at commit [`c9ff9a5`](https://github.com/apache/spark/commit/c9ff9a5536a3141baa785b87a7fce05543b90c84).
 * This patch merges cleanly.

tgaloppo commented:
@mengxr I was able to build the docs (I had to do a clean build on my source tree for some reason).  I have checked all links (that I added, anyway).  I also updated mllib-guide.md to include Gaussian mixture (and power iteration, since it was also missing) in the list of clustering techniques.

I've removed the WIP tag; I will not add any more content, unless requested.

SparkQA commented:
[Test build #26915 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/26915/consoleFull) for   PR 4401 at commit [`c9ff9a5`](https://github.com/apache/spark/commit/c9ff9a5536a3141baa785b87a7fce05543b90c84).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `public class GaussianMixtureExample `

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/26915/
Test PASSed.

mengxr commented:
LGTM. Merged into master and branch-1.3! Thanks for working on the user guide!
